C_Crud_Operation_in_File.c

```diff
@@ -0,0 +1,195 @@
+    scanf("%d", &newUser.age);
+
+    fprintf(file, "%d %s %d\n", newUser.id, newUser.name, newUser.age);
+    fclose(file);
+    printf("User add success\n");
+}
+
+// user read (print):
+void readUsers() {
+    FILE *file = fopen(FILE_NAME, "r");
+    if (!file) {
+        printf("Error occrured in opening file\n");
+        return;
+    }
+
+    char buffer[10000];
+    printf("All users are:\n");
+    while (fgets(buffer, sizeof(buffer), file) != NULL) { 
+        printf("%s", buffer); 
+    }
+
+    fclose(file);
+}
+
+// user update:
+void updateUser() {
+    int idToUpdate;
+    printf("Enter User ID to update: ");
+    scanf("%d", &idToUpdate);
+
+    FILE *file = fopen(FILE_NAME, "r");
+    if (!file) {
+        printf("Error occrured in opening file\n");
+        return;
+    }
+
+    FILE *tempFile = fopen("temp.txt", "w");
+    if (!tempFile) {
+        printf("Error occured in creating temporary file.\n");
+        fclose(file);
+        return;
+    }
+
+    int found = 0;
+    struct User user;
+    char buffer[10000]; 
+
+    while (fgets(buffer, sizeof(buffer), file) != NULL) { 
+        sscanf(buffer, "%d %99s %d", &user.id, user.name, &user.age); 
+        if (user.id == idToUpdate) { 
+            found = 1;
+            printf("Enter new Name: ");
+            getchar(); 
+            fgets(user.name, 100, stdin);
+            user.name[strcspn(user.name, "\n")] = '\0';  
+
+            printf("Enter new Age: ");
+            scanf("%d", &user.age);
+
+            fprintf(tempFile, "%d %s %d\n", user.id, user.name, user.age); 
+        } 
+        else fputs(buffer, tempFile); 
+    } 
+
+    fclose(file); 
+    fclose(tempFile); 
+    if (found) {
+        remove(FILE_NAME); 
+        rename("temp.txt", FILE_NAME); 
+        printf("User updated successfully.\n");
+    } 
+    else {
+        remove("temp.txt"); 
+        printf("User with ID %d not found.\n", idToUpdate);
+    }
+}
+
+// user delete:
+void deleteUser() {
+    int idToDelete;
+    printf("Enter User ID to delete: ");
+    scanf("%d", &idToDelete);
+
+    FILE *file = fopen(FILE_NAME, "r");
+    if (!file) {
+        printf("Error opening file.\n");
+        return;
+    }
+
+    FILE *tempFile = fopen("temp.txt", "w");
+    if (!tempFile) {
+        printf("Error creating temporary file.\n");
+        fclose(file);
+        return;
+    }
+
+    int found = 0;
+    struct User user;
+    char buffer[10000]; 
+
+    while (fgets(buffer, sizeof(buffer), file) != NULL) {
+        sscanf(buffer, "%d %99s %d", &user.id, &user.name, &user.age); 
+        if (user.id == idToDelete) found = 1;
+        else fputs(buffer, tempFile); 
+    }
+
+    fclose(file);
+    fclose(tempFile);
+
+    if (found) {
+        remove(FILE_NAME); 
+        rename("temp.txt", FILE_NAME); 
+        printf("User deleted successfully.\n");
+    } 
+    else {
+        remove("temp.txt"); 
+        printf("User with ID %d not found.\n", idToDelete);
+    }
+}
+
+void Operations() {
```

Comment:
Follow one naming conventions only. You are using camel case for funtions but here you used pascal case